signature/cose/envelope.go

```diff
@@ -631,20 +635,61 @@ func decodeTime(timeRaw cbor.RawMessage) (time.Time, error) {
 	return t, nil
 }
 
-// parseTime parses time values from cose.ProtectedHeader
-// in go-cose, CBOR times (tag 1) decode to time.Time.
+// parseTime validates Tag1 Datetime in cborMap given label then returns
+// time.Time value from cose.ProtectedHeader.
 //
 // For more details: https://github.com/fxamacker/cbor/blob/7704fa5efaf3ef4ac35aff38f50f6ff567793072/decode.go#L52
-func parseTime(timeValue interface{}) (time.Time, error) {
-	switch t := timeValue.(type) {
+func parseTime(cborMap map[interface{}]cbor.RawMessage, label string, protected cose.ProtectedHeader) (time.Time, error) {
+	err := validateTag(cborMap, label)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	switch t := protected[label].(type) {
 	case cbor.RawMessage:
 		return decodeTime(t)
-	// TODO: need a way to check the tag number of datetime inside the signature
-	// and fail if it's a Tag0. We only accept Tag1 datetime.
-	// https://github.com/notaryproject/notation-core-go/issues/97
 	case time.Time:
```

Comment:
Can we add a comment block stating that `time.Time` comes from the verification path?

Reply:
We only need to validate tag for verification path.

Reply:
Updated.